rustler_mix/lib/rustler/compiler/config.ex

```diff
@@ -0,0 +1,83 @@
+defmodule Rustler.Compiler.Config do
+  @moduledoc false
+
+  @type rust_version :: :stable | :beta | :nightly | binary()
+  @type cargo :: :system | {:rustup, rust_version()} | {:bin, Path.t()}
+  @type crate :: atom()
+  @type default_features :: boolean()
+  @type env :: [{binary(), binary()}]
+  @type features :: [binary()]
+  @type mode :: :debug | :release
+  @type load_data :: term()
+  @type path :: Path.t()
+
+  defstruct cargo: :system,
+            crate: nil,
+            default_features: true,
+            env: [],
+            external_resources: [],
+            features: [],
+            load_data: 0,
+            load_from: nil,
+            mode: :release,
+            otp_app: nil,
+            path: "",
+            priv_dir: "",
+            skip_compilation?: false,
+            target: nil,
+            target_dir: ""
+
+  alias Rustler.Compiler.Config
+
+  def from(otp_app, module, opts) do
+    config = Application.get_env(otp_app, module, [])
+
+    cargo = Keyword.get(config, :cargo, :system)
```

Comment:
With multiple NIFs present, this would allow each crate to be compiled with a different `:cargo` setting using the Application environment, right? If so, shouldn't `:cargo` be a global option?

Reply:
I'm not sure. I personally don't use this option and it has been here since before my time. I honestly would prefer that this wasn't an option (more for us to maintain) :smile:.